### PR TITLE
Allow compiling of SCSS Code

### PR DESCRIPTION
**Pull request by rndevuk**
This is part one, once this has been merged. I will create SCSS from the CSS in the current themes. this will serve 2 purposes.
1) Making updates to style information easier in the future
2) Serves as an example so that Designers know how to use the SCSS Feature.

### CODE REVIEW

File: src/bb-library/Box/TwigExtensions.php [hunk 1]
```diff
@@ -39,6 +39,7 @@ public function getDi()
     public function getFilters()
     {
         return array(
+            'compile_scss' => new Twig\TwigFilter('compile_scss', array($this, 'twig_compile_scss'), array('needs_environment' => false, 'is_safe' => array('html'))),
```
Comment (fordnox) [inline src/bb-library/Box/TwigExtensions.php:42]: rename to `scss`?

File: src/bb-library/Box/TwigExtensions.php [hunk 2]
```diff
@@ -79,6 +80,15 @@ public function getName()
     {
         return 'bb';
     }
+    
+    public function twig_compile_scss($file = 'style.scss', $forceCompile = false)
+    {
+        if (isset($_GET['SCSS_COMPILE']))
```
Comment (fordnox) [inline src/bb-library/Box/TwigExtensions.php:86]: this is a bad practice, and impossible to test.

**Comment by fordnox** [comment]
All scss compilation should be done in build process not in the code

**Comment by fordnox** [comment]
One more thing is that shopify did not introduce this filter https://shopify.dev/docs/themes/liquid/reference/filters

This filter will increase complexity of themes, and bad practices,css will be generated on the fly instead of once during the build process.

**Comment by rndevuk** [comment]
ok close this. I'll not waste any further time on it.

**Pull request closed without merging**